### ISSUE TITLE
some ts url with token is not inserted 

### ISSUE THREAD

**Issue by whatvn** [kaltura/nginx-secure-token-module#72]
Hi @erankor ,

Look like the bug this module has, config: 

```
secure_token $args;
```

Then sometime I had an output m3u8 with some ts url without token: 
```

ams5-n.com-299-v1-a1.ts?token=322424424
#EXTINF:9.000,
ams5-n.com-300-v1-a1.ts?token=322424424
#EXTINF:9.000,
ams5-n.com-301-v1-a1.ts
#EXTINF:9.000,
ams5-n.com-302-v1-a1.ts?token=322424424
#EXTINF:9.000,
.....
```

**Comment by erankor** [comment]
We are using it for years and never bumped into such a thing... 
I would need a way to reproduce it, maybe the full m3u8 & nginx conf.

**Comment by whatvn** [comment]
[index (43).m3u8.zip](https://github.com/kaltura/nginx-secure-token-module/files/2938416/index.43.m3u8.zip)

nginx config, i use map as you suggest, i think it's ok since token is inserted to the playlist, but some  missing, map.conf
```

map $http_x_tokenize $token {
       "no" "";
       default $args;
}
```

token config: 

```
# embed secure token into playlist
secure_token_types application/vnd.apple.mpegurl application/dash+xml text/xml;
secure_token $token;
```

that's it

**Comment by erankor** [comment]
Please reproduce it stand alone = a minimal nginx conf that serves an m3u8 from file and adds params to it, and provide the full nginx conf + input m3u8 (you sent the m3u8 after the embedding of the params).
I don't want to spend the time filling in the gaps from the details above, just to see it doesn't reproduce for me...

**Comment by whatvn** [comment]
hi @erankor, it's strange to me, since if I use token insert alone, without `map`, m3u8 file after token insertion is fine. 
If using map, with empty http header x-tokenize, segment 301 has no token insert, but it happen on random file, not all files have this problem. I dont know how to give you a concrete example

**Comment by whatvn** [comment]
i will try to create a minimal example which product the bug today

**Comment by Media182** [comment]
I'm having the same problem on one of my servers. 1-2 chunks out of 1000 do not have args at all.
CentOS 7.6, nginx 1.21.4, nginx-secure-token-module 1.5 latest version.
No other custom rewrites mods used.

**Comment by Media182** [comment]
I will try to reproduce and provide you SSH server details so you can check normally.
How I can send you server details securely?

**Comment by erankor** [comment]
You can email me at eran.kornblau at kaltura dot com

**Comment by erankor** [comment]
Fixed in #102, thanks @Media182!
